lib/active_merchant/billing/gateways/stripe_payment_intents.rb

```diff
@@ -418,74 +418,83 @@ def add_exemption(post, options = {})
+                                   return unless stored_credential[:initiator] == 'cardholder'
+
+                                   initial_transaction_stored_credential(post, stored_credential)
+                                 else
+                                   subsequent_transaction_stored_credential(post, stored_credential)
+                                 end
+
+        card_options = post[:payment_method_options][:card]
+        card_options[:stored_credential_transaction_type] = stored_credential_type
+        card_options[:mit_exemption].delete(:network_transaction_id) if stored_credential_type == 'setup_on_session'
       end
 
-      def initial_transaction_stored_credential(post, reason_type)
-        if reason_type == 'unscheduled'
+      def initial_transaction_stored_credential(post, stored_credential)
+        case stored_credential[:reason_type]
+        when 'unscheduled'
           # Charge on-session and store card for future one-off payment use
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'setup_off_session_unscheduled'
-        elsif reason_type == 'recurring'
+          'setup_off_session_unscheduled'
+        when 'recurring'
           # Charge on-session and store card for future recurring payment use
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'setup_off_session_recurring'
+          'setup_off_session_recurring'
         else
           # Charge on-session and store card for future on-session payment use.
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'setup_on_session'
+          'setup_on_session'
         end
       end
 
-      def subsequent_transaction_stored_credential(post, initiator, reason_type)
-        if initiator == 'cardholder'
+      def subsequent_transaction_stored_credential(post, stored_credential)
+        if stored_credential[:initiator] == 'cardholder'
           # Charge on-session customer using previously stored card.
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'stored_on_session'
-        elsif reason_type == 'recurring'
+          'stored_on_session'
+        elsif stored_credential[:reason_type] == 'recurring'
           # Charge off-session customer using previously stored card for recurring transaction
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'stored_off_session_recurring'
+          'stored_off_session_recurring'
         else
           # Charge off-session customer using previously stored card for one-off transaction
-          post[:payment_method_options][:card][:stored_credential_transaction_type] = 'stored_off_session_unscheduled'
+          'stored_off_session_unscheduled'
         end
       end
 
       def add_ntid(post, options = {})
         return unless options[:network_transaction_id]
+        return if options[:setup_future_usage] == 'off_session'
```

Comment:
This method is used when someone passes the NTID via the GSF override. I should move this check from `add_ntid` to `add_stored_credentials` so that if we send down `setup_future_usage ==  'off_session'` then it's a CIT and don't add any NTID flagging. What do you think?

Reply:
I think we should keep it as is for now. What if a customer just passes NTID but nothing else required by `add_stored_credentials`.